scenario_player/runner.py

```diff
@@ -153,289 +335,326 @@ def local_seed(self) -> str:
 
-        should_deposit_ud_token = udc_enabled and udc_settings.token.deposit
-        allowance_tx, required_allowance = self.udc.update_allowance()
-        if allowance_tx:
-            ud_token_tx.add(allowance_tx)
-        if should_deposit_ud_token:
+        return greenlets
 
-            tx = self.udc.mint(
-                our_address,
-                required_balance=required_allowance,
-                max_fund_amount=required_allowance * 2,
-            )
-            if tx:
-                ud_token_tx.add(tx)
+    def setup_token_contract_for_token_network(self, proxy_manager: ProxyManager) -> CustomToken:
+        """ Ensure there is a deployed token contract and return a `CustomToken`
+        proxy to it. This token will be used for the scenario's token network.
 
-        return ud_token_tx, udc_ctr, should_deposit_ud_token
+        This will either:
 
-    def _initialize_nodes(
-        self, nursery: Nursery
-    ) -> Tuple[Set[TransactionHash], gevent.Greenlet, Set[ChecksumAddress], int]:
-        """This methods starts all the Raiden nodes and makes sure that each
-        account has at least `NODE_ACCOUNT_BALANCE_MIN`.
+        - Use the token from the address provided in the scenario
+          configuration.
+        - Use a previously deployed token, with the details loaded from the
+          disk.
+        - Deploy a new token if neither of the above options is used.
         """
-        fund_tx: Set[TransactionHash] = set()
-
-        self.node_controller.initialize_nodes()
-        node_addresses = self.node_controller.addresses
-        node_count = len(self.node_controller)
-        balance_per_node = {
-            address: self.client.balance(to_canonical_address(address))
-            for address in node_addresses
-        }
-        low_balances = {
-            address: balance
-            for address, balance in balance_per_node.items()
-            if balance < NODE_ACCOUNT_BALANCE_MIN
-        }
-        log.debug("Node eth balances", balances=balance_per_node, low_balances=low_balances)
-        if low_balances:
-            log.info("Funding nodes", nodes=low_balances.keys())
-            for address, balance in low_balances.items():
-                params = {
-                    "client_id": self.definition.spaas.rpc.client_id,
-                    "to": address,
-                    "value": NODE_ACCOUNT_BALANCE_FUND - balance,
-                    "startgas": 21_000,
-                }
-                resp = self.service_session.post("spaas://rpc/transactions", json=params)
-                tx_hash = resp.json()["tx_hash"]
-                fund_tx.add(tx_hash)
+        token_definition = self.definition.token
+        reuse_token_from_file = token_definition.reuse_token
+        token_address = to_canonical_address(token_definition.address)
+        deploy_new = not (reuse_token_from_file or token_address)
+        token_data_path = str(self.data_path.joinpath("token.info"))
```

Comment:
Why not keep this a Path instance?

Reply:
It turned out to be less code with the same functionality.

Reply:
Hm ok, not important.